ledger/conway/conway.go

```diff
@@ -601,16 +607,52 @@ type ConwayTransaction struct {
 }
 
 func (t *ConwayTransaction) UnmarshalCBOR(cborData []byte) error {
-	type tConwayTransaction ConwayTransaction
-	var tmp tConwayTransaction
-	if _, err := cbor.Decode(cborData, &tmp); err != nil {
+	// Decode as raw array to preserve metadata bytes
+	var txArray []cbor.RawMessage
+	if _, err := cbor.Decode(cborData, &txArray); err != nil {
 		return err
 	}
-	*t = ConwayTransaction(tmp)
+
+	// Ensure we have at least 3 components
+	if len(txArray) < 3 {
+		return fmt.Errorf(
+			"invalid transaction: expected at least 3 components, got %d",
+			len(txArray),
+		)
+	}
+
+	// Decode body
+	if _, err := cbor.Decode([]byte(txArray[0]), &t.Body); err != nil {
+		return fmt.Errorf("failed to decode transaction body: %w", err)
+	}
+
+	// Decode witness set
+	if _, err := cbor.Decode([]byte(txArray[1]), &t.WitnessSet); err != nil {
+		return fmt.Errorf("failed to decode transaction witness set: %w", err)
+	}
+
+	// Decode TxIsValid flag
+	if _, err := cbor.Decode([]byte(txArray[2]), &t.TxIsValid); err != nil {
+		return fmt.Errorf("failed to decode TxIsValid: %w", err)
+	}
+
+	// Handle metadata (component 4)
+	// DecodeAuxiliaryDataToMetadata already preserves raw bytes via DecodeMetadatumRaw
+	if len(txArray) > 3 && len(txArray[3]) > 0 {
```

Comment:
The TX length is not variable. The last element will always exist, and will either contain metadata or a, CBOR `nil`